tools/_functions.sh

```diff
@@ -473,3 +473,24 @@ function join_by {
         printf %s "$f" "${@/#/$d}"
     fi
 }
+
+# This function checks if the flag "--as-precommit" was set, and if so,
+# runs the specified pre-commit hook against all changed files
+function run_as_precommit {
+    local command="$1"
+    shift
+
+    for arg in "$@"; do
+        if [[ "$arg" == "--as-precommit" ]]; then
+            FILES=$(git diff --cached --name-only --diff-filter=ACMR | sed 's| |\\ |g')
```

Comment:
I don't fully understand why we need the `| sed 's| |\\ |g'` part, but this escaping causes some problems when we have files with spaces:
![image](https://github.com/digitalfabrik/integreat-cms/assets/78504586/fbfc3aa2-9a67-4b89-955d-61d675c60324)

Without this part, the file can be found again

Reply:
I think this is a mistake in the original script, you are right. Pretty sure it should just be `| sed 's| |\ |g'`.